schema/binary.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Namespace Labs Inc; All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+
+package schema
+
+import (
+	"google.golang.org/protobuf/proto"
+	"namespacelabs.dev/foundation/internal/fnerrors"
+)
+
+// Modifies the instance in place.
+func (bc *BinaryConfig) MergeFrom(other *BinaryConfig) error {
+	if other == nil {
+		return nil
+	}
+
+	if other.Command != nil {
+		bc.Command = other.Command
+	}
+	if other.WorkingDir != "" {
+		bc.WorkingDir = other.WorkingDir
+	}
+
+	// TODO: consider if this should be a replacement.
+	bc.Args = append(bc.Args, other.Args...)
+
+	var err error
+	bc.Env, err = mergeEnvs(bc.Env, other.Env)
+	return err
+}
+
+// Returns a deep copy with args added to the end of the list.
+func (bc *BinaryConfig) WithArgs(args ...string) *BinaryConfig {
```

Comment:
I see the usefulness of this method, but similarly I think its best to keep it out of protos.